### ISSUE TITLE
Last Boss Graphics Problem (Contra Shattered Soldier) 

### ISSUE THREAD

**Issue by ReyVGM** [PCSX2/pcsx2#1711]
I don't know if this is a problem or not, but it certainly doesn't look right.

While fighting the true last boss, fire/explosion comes out from the right of the screen and as you can see, it doesn't look right. I checked a youtube video (presumably from a real PS2) and it shows the fire correctly.
https://youtu.be/vg9QYmnfEyc?t=2258

![gsdx_20161218013321](https://cloud.githubusercontent.com/assets/7338867/21291753/67860ed0-c4c2-11e6-8eee-85ffcc3f79fb.png)

From the video, it looks as if the real PS2 just cuts that part out, but on the emulator it shows the whole thing?

**Comment by Hirato** [comment]
There's stuff on the left that isn't in the video either; are you using widescreen hacks?

**Comment by ReyVGM** [comment]
Nope.
All settings are at default and screenshot was taken at native resolution.

**Comment by ReyVGM** [comment]
If you don't get "S Rank" the true last boss above doesn't happen, but the fire is there and it appears correctly.
![gsdx_20161218125712](https://cloud.githubusercontent.com/assets/7338867/21295002/9c8873a2-c521-11e6-87e9-aae6cbe195b6.png)

**Comment by willkuer** [comment]
The images you show are clearly taken with widescreen aspect ratio. Ws hacks are enabled by default. It is highly probable the widescreen hack that shows off-screen stuff. Switch to 4:3 aspect ratio in emulation settings -> gs window, or disable widescreen hacks in the top menu. The problem should be gone.

**Comment by ReyVGM** [comment]
I understand, but I don't have Widescreen enabled. I am taking a native screenshot, so changing to widescreen or 4:3 doesn't change the way the screenshot comes out because I'm not taking a screenshot of the screen, but (allegedly) directly from the framebuffer. I also have 4:3 selected in the PS2 bios.
And even changing between 4:3, widescreen, fit to screen, or anything else still shows that "problem" because all that does is stretch or squish what's on the screen.

Either it's an emulation problem, or the actual game is like that, BUT a real PS2 just cut parts of the left and right sending the signal to a TV.

![screen0846](https://cloud.githubusercontent.com/assets/7338867/21295437/70eede52-c52b-11e6-823c-081c0cfd74fe.png)
![screen0847](https://cloud.githubusercontent.com/assets/7338867/21295438/70f37a48-c52b-11e6-99e7-1bea1047d69c.png)

**Comment by FlatOutPS2** [comment]
But is `Enable Cheats` or `Enable Widescreen Patches` enabled on the PCSX2 System tab?

**Comment by willkuer** [comment]
Hm... on my mobile it looked like widescreen aspect ratio. Here on my pc (and reading the image information) it is obvious that your screenshots are in 4:3. Probably some illusion due to the mobile screen aspect ratio.

I revoke my claim. It's probably not due to ws hacks.

**Comment by FlatOutPS2** [comment]
> Hm... on my mobile it looked like widescreen aspect ratio. Here on my pc (and reading the image information) it is obvious that your screenshots are in 4:3. Probably some illusion due to the mobile screen aspect ratio.
I revoke my claim. It's probably not due to ws hacks.

That doesn't mean anything. Widescreen hacks don't affect PCSX2 aspect ratio or image information/resolution.

**Comment by willkuer** [comment]
But they should not be active. (Except as you said when they are in the cheats folder)

**Comment by FlatOutPS2** [comment]
> But they should not be active. 

Why not?

**Comment by ReyVGM** [comment]
"But is Enable Cheats or Enable Widescreen Patches enabled on the PCSX2 System tab?"

Nope. I don't have cheats or Widescreen enabled on the system menu tab.

It's possible, and bear with me because I don't know how PS2 works, but it's possible it's doing something the N64 did too?
The N64 had the info coming from the framebuffer, but before the signal went out to the TV, it first passed through the VI filters, which gave it the trademark N64 'blur'.
Now, on Turok 2, if you took a native res screen before it went through the VI filters, the image would come out full resolution. But if you took the screenshot after it passed through the VI filters, the image would be enclosed in borders (probably to achieve a better framerate). However, those borders could not be seen on a CRT due to overscan.

Judging from that youtube video, it seems the real PS2 is cutting off information from the left and right sides. Information the emulator is showing, which is completely normal.

The point of this thread was to find out if it's a mistake in emulation and the fire is indeed showing up incorrectly, or if that's how the game really renders that scene BUT for some reason when the image is sent to a TV through a real PS2, then it gets cut off by borders or just showing up out of bounds.

**Comment by FlatOutPS2** [comment]
Have you tried the game on your PS2 to make sure it's not a bug in the game(game on youtube might be a different version/region)?

**Comment by ReyVGM** [comment]
Never could reach the true last boss on a real PS2. You have to S rank the entire game. I did it on emulation thanks to save states.
I've checked other videos and they all show the cropped visuals. My guess is the emulator is showing everything from the framebuffer and the real PS2 did something to hide it.

**Comment by FlatOutPS2** [comment]
As the snapshot posted earlier shows you using Direct3D11 (Hardware) as Renderer, did you also try "OpenGL (Hardware)" or switching to Software mode(F9 key)?

**Comment by ReyVGM** [comment]
"Fixed" it. It was a D3D11 issue. I switched to D3D9 and it looks fine now. But there seems to be graphical glitches now, like that black line in the plane. Could it be because I'm lacking a video card?

![ascreen0857](https://cloud.githubusercontent.com/assets/7338867/21297806/0fe0641e-c55d-11e6-8baa-be1f6b3170f8.png)
![gsdx_20161218200004](https://cloud.githubusercontent.com/assets/7338867/21297807/0fe1fcfc-c55d-11e6-8b50-6811006280a0.png)

**Comment by willkuer** [comment]
Opengl (hw)? Dx11 (sw)?

**Comment by ReyVGM** [comment]
OpenGl HW crashes the emu for me.

**Comment by ReyVGM** [comment]
Which renderer is supposed to be closer to the real hardware?  A soft renderer or a hardware one? 

I noticed that using a software renderer textures look grittier and using a hardware one textures look cleaner.

**Comment by FlatOutPS2** [comment]
Can you upload a GS dump of the issue when it's visible? http://forums.pcsx2.net/Thread-How-to-create-a-proper-GS-dump

> Which renderer is supposed to be closer to the real hardware? A soft renderer or a hardware one?
I noticed that using a software renderer textures look grittier and using a hardware one textures look cleaner.

Software mode is closer to the real hardware. The game might have a filter that makes it look grittier, which isn't properly emulated in hardware mode.

**Comment by ReyVGM** [comment]
@FlatOutPS2 

I don't understand what's going on now.
I got to the point where the issue happens (without using a save state, to avoid renderer conflicts) and now it doesn't happen while using D3D11 (hw) like it did before.  Does the emu save preferences per game or dumps something from the game that is "saving" whatever change that made things right the last time?
Maybe I could delete whatever that may be and try again. But as of right now, I'm not getting the graphical error talked before, and I haven't changed any options or downloaded a new emu version.

**Comment by ReyVGM** [comment]
I think I figured out what the whole problem was.

When I initially started playing Contra Shattered Solder, I did have the widescreen hack cheat code enabled without knowing about it (since the emulator doesn't have a dialog to see what cheats are on/off). I then disabled the cheats, but since I had already made a save state, I continued with the effects on without knowing about it.

So, while the screenshot's resolution was the native one, and I had 4:3 mode enabled, the graphics were indeed in widescreen mode which caused the game to show "out of bounds" stuff like that missing fire on the top screenshot.

So in the end, it wasn't an emulation issue. I deleted all the widescreen hack cheats since I won't use them and I don't want something like this to happen again.

This thread can be closed and thanks to everyone that tried to figure out what was going on.
